src/types/interface.ts

```diff
@@ -120,17 +168,98 @@ export class InterfaceType<Props extends Properties, ResultType>
+        const [opts, props] = decodeOptionalOptions<InterfaceMergeOptions, OtherProps>(args);
+        return this._mergeWith(opts, toPropsInfo(props, false), [], false, 'withRequired');
+    }
+
+    /**
+     * Create a new type by merging all properties of the given type into the properties of this type.
+     *
+     * @remarks
+     * Keys of the given (right hand side) type override keys of this type (left hand side). Any options such as `strictMissingKeys` and
+     * custom names are inherited from the left hand side type, but custom parsers and validators are dropped.
+     */
+    mergeWith<OtherProps extends Properties, OtherType>(
+        ...args:
+            | [type: InterfaceType<OtherProps, OtherType>]
+            | [name: string, type: InterfaceType<OtherProps, OtherType>]
+            | [options: InterfaceMergeOptions, type: InterfaceType<OtherProps, OtherType>]
+    ): MergeType<Props, ResultType, OtherProps, OtherType> {
+        const [opts, other] = decodeOptionalOptions<InterfaceMergeOptions, InterfaceType<OtherProps, OtherType>>(args);
+        return this._mergeWith(opts, other.propsInfo, other.customValidators, !!other.typeParser, 'mergeWith');
+    }
+
+    private _mergeWith<OtherProps extends Properties, OtherType>(
+        { name = this.isDefaultName ? undefined : this.name, ignoreParsers, ignoreValidations }: Partial<InterfaceMergeOptions>,
+        otherPropsInfo: PropertiesInfo<OtherProps>,
+        otherCustomValidators: ReadonlyArray<
+            <T extends ResultType & OtherType>(this: void, input: T, options: ValidationOptions) => Result<T>
+        >,
+        otherHasCustomParser: boolean,
+        method: string,
+    ): MergeType<Props, ResultType, OtherProps, OtherType> {
+        const customValidators = ignoreValidations ? [] : [...this.customValidators, ...otherCustomValidators];
+        if (customValidators.length) {
+            // Check that we have no conflicting properties. If there are no conflicting properties then each validator is still safe to
+            // run, each type in a TypeScript intersection is assignable to each of the elements in the intersection.
+            const conflictingProperties = Object.keys(otherPropsInfo).filter(key => key in this.propsInfo);
+            if (conflictingProperties.length) {
+                const conflicts = humanList(conflictingProperties, 'and', key => `<${printKey(key)}>`);
+                const propertiesAre = plural(conflictingProperties, 'property is', 'properties are');
+                throw new Error(
+                    `Error in ${this.name}.${method}(): Merge operation not allowed because one of the types has custom ` +
+                        `validations applied and the following ${propertiesAre} defined on both sides: ${conflicts}, ` +
+                        'use `ignoreValidations` to suppress this error or remove the conflicting property.',
```

Comment:
Don't forget to change this as well, if we alter the documentation/name of `ignoreValidations`.